src/app/cluster/details/external-cluster/template.html

```diff
@@ -56,7 +56,7 @@
         <i class="km-icon-mask km-icon-download"></i>
         <span>Get Kubeconfig</span>
       </button>
-      <button *ngIf="showAddMachineDeploymentButton"
+      <button *ngIf="!cluster.cloud.bringYourOwn"
```

Comment:
I think `bringYourOwn` property does not exist in `cluster.cloud`? Can you please double check?

Reply:
it's been add recently and the `bringYourOwn` object come from the backend empty so this is only to check if it's custom  external cluster 
![Screenshot from 2022-08-18 01-16-22](https://user-images.githubusercontent.com/85109141/185253260-44cdd5bf-90dd-41e6-b56a-9835362cb701.png)

Reply:
Then I think we should also add it in `ExternalCloudSpec` class. Wdyt?

```
export class ExternalCloudSpec {
  aks?: AKSCloudSpec;
  eks?: EKSCloudSpec;
  gke?: GKECloudSpec;
}
```

cc @KhizerRehan

Reply:
actually it's not been used in any place in the code but here and it's an empty object so i assume no need to add it right now in here

Reply:
We might get an error that this property doesn't exist in `ExternalCloudSpec` so better safe than sorry ;)

Reply:
so i just add it like `bringYourOwn : {}` or  `bringYourOwn : object` ? something like that inside  `ExternalCloudSpec`

Reply:
You can take a look at `CloudSpec` class in `src/app/shared/entity/cluster.ts`

Reply:
@ahmadhamzh better to add property + add a small comment too `bringYourOwn` checks only if cluster is custom  for future reference?

Reply:
No need for comment I think because `bringYourOwn` is self explanatory